src/compiler/crystal/codegen/exception.cr

```diff
@@ -4,197 +4,222 @@ class Crystal::CodeGenVisitor
   @node_ensure_exception_handlers = {} of UInt64 => Handler
 
   def visit(node : ExceptionHandler)
+    windows = @program.has_flag? "windows"
+
+    context.fun.personality_function = @llvm_mod.functions[@personality_name] if windows
+
+    # This is the block which is entered when the body raises an exception
     rescue_block = new_block "rescue"
 
     node_rescues = node.rescues
     node_ensure = node.ensure
     node_else = node.else
     rescue_ensure_block = nil
 
-    # Here we transform this:
```

Comment:
I would say that this whole comment section should stay. Your additions inlines are great, but keeping this sections with numbers and later references would be ideal. They facilitate understanding the later code a lot. 

Other than that LGTM.

Reply:
I don't like that block comment because it's very incomplete information, and once you distill it into it's core it's just what I've written below.

Once I start thinking how to improve the comment, I just end up with repeating what i've written below

Reply:
The *once you distill* time is decreased a lot with a nice introduction like the one that is removed. I think that the intricate nature of the code transformation deserves that kind of introduction. Just restoring those line are enough.